### ISSUE TITLE
Find API calls recipe returns test results

### ISSUE THREAD

**Issue by mike-solomon** [openrewrite/rewrite-spring#428]
In Moderne, when I run the [Find API calls recipe](https://app.moderne.io/recipes/org.openrewrite.java.spring.search.FindApiCalls) on the `Default` organization, I notice that most of the results are tests rather than actual API calls:

![image](https://github.com/openrewrite/rewrite-spring/assets/1187834/47ff778f-cbba-42e6-901e-d74db5921280)

[Link to the test code](https://github.com/spring-projects/spring-credhub/blob/fe80465f09e15dbb4b34019b361477a2c31c74fc/spring-credhub-core/src/test/java/org/springframework/credhub/core/permission/CredHubPermissionTemplateUnitTests.java#L63-L78)

In looking into those test files, I see that the calls are being mocked. I'm not sure how feasible it would be, but can this recipe be tuned to not return results like that? Or am I perhaps misunderstanding something? 

I would expect this recipe to return external API calls and not test/mocked ones.

**Comment by knutwannheden** [comment]
@mike-solomon Yes, I agree. We could probably use the `JavaSourceSet` to figure this out. Unfortunately we don't (yet) have a `test` property on `SourceSet`. I think we have up until now just been checking the name of the source set. But with Gradle and Bazel the source set names can practically be anything, so having a property would make more sense, IMHO.

**Comment by knutwannheden** [comment]
I added a feature request for this: https://github.com/openrewrite/rewrite/issues/3512
